### PR TITLE
feat(Wallet): cache fetched balance history to DB for efficiency

### PR DESCRIPTION
**Pull request by stefandunca**
### Implements: #8226

Changes:

- Start balance history fetching "service" on the start
- Add helper tests to C++ to validate startup
- Bumps [status-go PR](https://github.com/status-im/status-go/pull/2962)
- Small fixes and improvements

### CODE REVIEW

File: libs/StatusGoQt/src/StatusGo/Wallet/WalletApi.cpp [hunk 1]
```diff
@@ -109,9 +109,9 @@ TokenBalances getTokensBalancesForChainIDs(const std::vector<ChainID>& chainIds,
 }
 
 std::vector<TokenBalanceHistory>
-getBalanceHistory(const ChainID& chainID, Accounts::EOAddress account, BalanceHistoryTimeInterval timeInterval)
+getBalanceHistory(const ChainID& chainID, Accounts::EOAddress account, QString currency, BalanceHistoryTimeInterval timeInterval)
```
Comment (endulab) [inline libs/StatusGoQt/src/StatusGo/Wallet/WalletApi.cpp:112]: const QString &

File: libs/StatusGoQt/src/StatusGo/Wallet/WalletApi.h [hunk 2]
```diff
@@ -87,10 +87,23 @@ enum BalanceHistoryTimeInterval
 /// \note status-go's API -> GetBalanceHistory@api.go
 /// \throws \c CallPrivateRpcError
 std::vector<TokenBalanceHistory>
-getBalanceHistory(const ChainID& chainID, Accounts::EOAddress account, BalanceHistoryTimeInterval timeInterval);
+getBalanceHistory(const ChainID& chainID, Accounts::EOAddress account, QString currency, BalanceHistoryTimeInterval timeInterval);
```
Comment (endulab) [inline libs/StatusGoQt/src/StatusGo/Wallet/WalletApi.h:90]: const QString &

File: ui/imports/shared/stores/ChartStoreBase.qml [hunk 3]
```diff
@@ -14,7 +14,7 @@ Item {
         All
     }
 
-    readonly property var timeRangeTabsModel: [
+    property var timeRangeTabsModel: [
```
Comment (endulab) [inline ui/imports/shared/stores/ChartStoreBase.qml:17]: why not readonly ?

**Comment by anastasiyaig** [comment]
@stefandunca could u please resolve conflicts? Sorry about that :( 
@Khushboo-dev-cpp @endulab @alaibe please review

**Review by saledjenic — APPROVED** [review]

**Pull request closed without merging**